### ISSUE TITLE
Fail to disable displays on Win11 24H2

### ISSUE THREAD

**Issue by mikef0** [Cynary/sunshine-virtual-monitor#16]
Hi. I was previously using the script without issues on my Windows 10 machine. I recently switched to a new PC running the latest version of Windows 11 and get the following error.

I have tried different resolutions, HDR and non HDR versions of the virtual display driver. Nothing seems to work.

Setting up a moonlight monitor with 3840x2160@60 with hdr on
Enabling the virtual display.
Setting up the virtual display and disabling other displays.
Disabling all other displays.
Failed to disable all other displays.
At C:\Sunshine_Scripts\tools\sunshine-virtual-monitor-main\setup_sunvdm.ps1:109 char:9
+         Throw "Failed to disable all other displays."
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Failed to disable all other displays.:String) [], RuntimeException
    + FullyQualifiedErrorId : Failed to disable all other displays.

**Comment by mikef0** [comment]
I should have noted. The virtual display driver works fine if I don't use the script. There are no errors or anything of interest in the Sunshine logs except simply informing me that it is running your start and termination scripts as expected.

**Comment by dismantlethedictator** [comment]
yes - same issue on my end. I recently reinstalled windows and the scripts are popping an identical error in the log. replicated the error with multiple clients. Was able to stream to clients once i deleted the sunshine scripts

**Comment by Cynary** [comment]
Hi,
Thanks for the report - this seems really similar to #9 
Can you confirm that you were running the most recent version of the scripts?

Also, can you share the output of the following when the virtual display is enabled:
```Get-PnpDevice -Class Display
```

Thanks!

**Comment by dismantlethedictator** [comment]
yes running the most recent version of the scripts and just tried the fix from the thread you linked and it doesnt seem to help. sorry what output specifically?

**Comment by Cynary** [comment]
HI @dismantlethedictator I messed up the formatting in my previous comment; I have fixed it now - please run the command:
```
Get-PnpDevice -Class Display
```

Thanks!

**Comment by mikef0** [comment]
Please see below.

Status     Class           FriendlyName                                                                     InstanceId
------     -----           ------------                                                                     ----------
OK         Display         NVIDIA GeForce GTX 1070                                                          PCI\VEN_10DE...
OK         Display         Virtual Display with HDR                                                         ROOT\DISPLAY...
Unknown    Display         Microsoft Remote Display Adapter                                                 SWD\REMOTEDI...

**Comment by dismantlethedictator** [comment]
> HI @dismantlethedictator I messed up the formatting in my previous comment; I have fixed it now - please run the command:
> 
> ```
> Get-PnpDevice -Class Display
> ```
> 
> Thanks!


Status     Class           FriendlyName                                                                     InstanceId
------     -----           ------------                                                                     ----------
OK         Display         IddSampleDriver Device HDR                                                       ROOT\DIS...
OK         Display         AMD Radeon RX 6700 XT                                                            PCI\VEN_...

**Comment by dismantlethedictator** [comment]
```
Get-PnpDevice -Class Display | Where-Object {$_.FriendlyName -like "*idd*" -or $_.FriendlyName -like "*mtt*"}

Status     Class           FriendlyName                                                                     InstanceId
------     -----           ------------                                                                     ----------
OK         Display         IddSampleDriver Device HDR                                                       ROOT\DIS...
```

as a heads up, cmd is able to locate my virtual display using friendlyname like "idd" or "mrt"

**Comment by Cynary** [comment]
Yeah, this is likely some different issue.
Would you be able to help debug by running the script one line at a time, and reporting:
- Does it turn on the virtual display correctly, and set the resolution, etc.?
- Does it emit any other errors when you actually run it?

A few things it could be:
`WindowsDisplayManager` may have some issue, and is unable to delete the display.
We're not properly enabling the virtual display.

Not sure what else, but running the script one line at a time, and checking the status of displays etc. should hopefully help.

**Comment by mikef0** [comment]
It never disables the virtual monitor. I have seen it randomly behave in 2 different ways and I can't account for what causes the difference in behavior.

1. I try to connect with sunshine -> My SECONDARY monitor goes blank, virtual monitor stays disabled. (sunshine times out and is no longer part of the equation. From this point, my secondary monitor is still disabled and I can't even see it in windows display manager to turn it on. If I ENABLE the virtual monitor, my secondary monitor comes back on but if I then disable the virtual monitor, I lose my secondary monitor again. I have to restart to get my secondary monitor back

2. Similar to above, except my secondary monitor position in relation to my primary monitor changes. Sunshine times out, virtual monitor is still disabled and I have to go in windows display manager to fix the position of my secondary monitor.

**Comment by Cynary** [comment]
I'm surprised by what you're reporting.
@mikef0 Just to confirm - are you running the most recent version? There was an issue in #9 just like yours that was caused by the different name of your virtual display.
Are you able to run the steps on the script 1 by 1? Also, when you say you are enabling/disabling the virtual monitor, what do you mean by that, exactly? Are you doing it via the display settings page, or are you going into the device manager (This is more like what the script is doing)? I wonder if the remote display you have setup could be causing some of the issues you're seeing. Could you share more details about that?

For @mikef0 and @dismantlethedictator - are you able to run the commands on the script 1 by 1 and report the state that you see? Something is going wrong given what you're reporting, but the logging isn't catching anything, so it's hard to tell :/

**Comment by dismantlethedictator** [comment]
i was able to fix the issue by doing the following steps:

1) Uninstall and delete all previous instances of sunshine, virtual display drivers and any custom scripts
2) prior to fresh installing the virtualdisplaydriver, i added my custom resolutions to "option.txt* and installed the driver
3) install sunshine, install scripts. working as it should now.

P.S - i download and install the virtual driver exclusively on the C drive, not sure if that helps smooth out the process.

**Comment by LeGeRyChEeSe** [comment]
Bonjour, je vous donne un feedback récent sur un problème similaire, j'ai eu le même problème mais sur un PC portable. Le PC freeze pendant 5 minutes, et l'écran clignote sans que je ne puisse rien faire d'autre qu'attendre. La connexion via moonlight tourne en boucle également jusqu'à échouer.

**Comment by chewie48** [comment]
Hi! I was having the same issue. The WindowsDisplayManager module wasn't working because the line `SetExecutionPolicy -Scope Process -ExecutionPolicy Bypass` is placed after the `Import-Module WindowsDisplayManager`

**Comment by marian-code** [comment]
I hit tha same problem a few days ago. I tried to debug it and narrowed it down to MultiMonitorTool not being able to change the primary display. SOmetimes randomly it does work but most of the time The command just doe not do anything. There is no error message or anything else. It fails silently.
